jxl/src/render/stages/upsample.rs

```diff
@@ -140,88 +86,40 @@ impl RenderPipelineStage for VerticalChromaUpsample {
 mod test {
     use super::*;
     use crate::{error::Result, image::Image, render::test::make_and_run_simple_pipeline};
-    use rand::SeedableRng;
     use test_log::test;
 
     #[test]
-    fn nn_consistency() -> Result<()> {
-        crate::render::test::test_stage_consistency::<_, u8, u8>(
-            NearestNeighbourUpsample::new(0),
-            (500, 500),
-            1,
-        )
-    }
-
-    #[test]
-    fn test_nn() -> Result<()> {
-        let image_size = (500, 400);
-        let input_size = (image_size.0 / 2, image_size.1 / 2);
-        let mut rng = rand_xorshift::XorShiftRng::seed_from_u64(0);
-        let input = vec![Image::<u8>::new_random(input_size, &mut rng)?];
-        let stage = NearestNeighbourUpsample::new(0);
-        let output: Vec<Image<u8>> =
-            make_and_run_simple_pipeline(stage, &input, image_size, 256)?.1;
-        assert_eq!(image_size, output[0].size());
-        for y in 0..image_size.1 {
-            for x in 0..image_size.0 {
-                let ix = x / 2;
-                let iy = y / 2;
-                let i = input[0].as_rect().row(iy)[ix];
-                let o = output[0].as_rect().row(y)[x];
-                if i != o {
-                    panic!("Mismatch at output position {x}x{y}: {i} vs output {o}");
-                }
-            }
-        }
-
-        Ok(())
-    }
-
-    #[test]
-    fn hchr_consistency() -> Result<()> {
-        crate::render::test::test_stage_consistency::<_, f32, f32>(
-            HorizontalChromaUpsample::new(0),
-            (500, 500),
-            1,
-        )
-    }
-
-    #[test]
-    fn test_hchr() -> Result<()> {
-        let mut input = Image::new((3, 1))?;
+    fn test_upsample2() -> Result<()> {
```

Comment:
Can we also test:
- consistency
- upsampling a constant image
?

Reply:
yes, added those